.github/workflows/code_checks.yml

```diff
@@ -1,5 +1,11 @@
 name: Code Checks
-on: [push]
+on:
+  pull_request:
+    branches:
```

Comment:
Do we want this to run on all branches, not just master? e.g. someone opens a PR against a branch that isn't master.